### DISCUSSION TITLE
Model rotates incorrectly

### DISCUSSION THREAD

**Discussion by DKGH** [google/model-viewer#4426] [Q&A]
I am using the below code:

```
<model-viewer src={urlToBlob} disable-tap camera-controls touch-action />
```

However, it starts to tumble when I try to rotate the view left or right. Rotating the view top to bottom works fine. How do I ensure that left to right rotation also works correctly? Also, the up and down rotation gets stuck after a point. It seems constrained. How do get rid of the constrains. Is there a way to set orientation/view by clicking a button?

- model-viewer: v1.11.1

### Browser Affected
- [ ] Chrome

**Comment by elalish** [comment]
You have a bunch of different questions here. `touch-action` is not a boolean - not sure what you're hoping for there. Yes, rotation is constrained to keeping the up direction up (for UX reasons). The constraints are adjustable - see the docs. Yes, you can set the camera by clicking a button: `mv.cameraOrbit="..."`, but of course you'll have to do the math.
